packages/@vuepress/bundler-vite/src/plugin/createConstantReplacementPlugin.ts

```diff
@@ -0,0 +1,53 @@
+import type { Plugin } from 'vite'
+
+const fileRegex = /\.html\.vue$/
+const templateRegex = /<template>[\s\S]*<\/template>/g
+
+export const createConstantReplacementPlugin = (): Plugin => {
+  let defineRegex: RegExp | null = null
+
+  return {
+    name: 'vuepress:constant-replacement',
+    enforce: 'pre',
+    apply: 'build',
+    configResolved(resolvedConfig) {
+      if (resolvedConfig.define) {
+        const define = Reflect.ownKeys(resolvedConfig.define) as string[]
+        defineRegex = new RegExp(
+          `\\b(${define
+            .map((key) => key.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&'))
+            .join('|')})`,
+          'g'
+        )
+      }
+    },
+    transform(code, id) {
+      if (fileRegex.test(id)) {
+        for (const match of code.matchAll(templateRegex)) {
+          let html = match[0]
+
+          // avoid env variables being replaced by vite
+          html = html
+            .replace(/\bimport\.meta/g, 'import.<wbr/>meta')
+            .replace(/\bprocess\.env/g, 'process.<wbr/>env')
+
+          // also avoid replacing vite user defines
+          if (defineRegex) {
+            html = html.replace(
+              defineRegex,
+              (_) => `${_[0]}<wbr/>${_.slice(1)}`
+            )
+          }
+
+          const index = match.index!
+          code =
+            code.substring(0, index) +
+            html +
+            code.substring(index + html.length)
```

Comment:
I'm afraid that the index would be wrong in the next loop if you change the `code` this way.

Reply:
I did not fully understand what you mean, each file will only be transformed once. Will this problem still occur after 27b9f02 that I had remove the loop.

Reply:
Hmmm, maybe my previous comment is not fully correct.

The `html.length` here is already different from the original `match[0].length`

I wonder if you could get correct result here. 🤔

Reply:
Fixed! my typo here.

Reply:
As for my first comment:

The `code` length is already changed, then the `match.index` in next loop would be incorrect.

Of course it won't happen after removing the loop.

Reply:
Incredible! I never realized it was a bug, thanks for your careful consideration and points out ❤️